src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -680,10 +685,22 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return undefined;
 	}
 
+	private getCellAttachmentFromSource(source: string | string[], attachments: nb.ICellAttachments): nb.ICellAttachments {
+		let newCellAttachment = {};
+		let newSource = Array.isArray(source) ? source.join() : source;
+		for (let key of Object.keys(attachments)) {
+			if (newSource.includes(`![${key}](attachment:${key})`)) {
```

Comment:
It would be preferable to use the same regex that https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/services/notebook/browser/models/cell.ts#L337 uses